tests/unit/admin/test-class-wc-rest-payments-settings-controller.php

```diff
@@ -548,10 +548,122 @@ public function account_business_support_address_validation_provider() {
 				],
 				$request,
 				'account_business_support_address',
-				new WP_Error( 'rest_invalid_pattern', 'Invalid address format!' ),
+				new WP_Error( 'rest_invalid_pattern', 'Error: Invalid address format!' ),
 			],
 		];
 	}
 
+	/**
+	 * Tests account business support email validator
+	 *
+	 * @dataProvider account_business_support_email_validation_provider
+	 */
+	public function test_validate_business_support_email( $value, $request, $param, $expected ) {
+		$return = $this->controller->validate_business_support_email_address( $value, $request, $param );
+		$this->assertEquals( $return, $expected );
+	}
+
+	/**
+	 * Provider for test_validate_business_support_email.
+	 * @return array[] test method params.
+	 */
+	public function account_business_support_email_validation_provider() {
+		$request = new WP_REST_Request();
+		return [
+			[
+				'test@test.com',
+				$request,
+				'account_business_support_email',
+				true,
+			],
+			[
+				'', // Empty value should trigger error.
+				$request,
+				'account_business_support_email',
+				new WP_Error( 'rest_invalid_pattern', 'Error: Support email address is required!' ),
+			],
+			[
+				'test@test',
+				$request,
+				'account_business_support_email',
+				new WP_Error( 'rest_invalid_pattern', 'Error: Invalid email address: test@test' ),
+			],
+		];
+	}
+
+	/**
+	 * Tests account business support phone validator
+	 *
+	 * @dataProvider account_business_support_phone_validation_provider
+	 */
+	public function test_validate_business_support_phone( $value, $request, $param, $expected ) {
+		$return = $this->controller->validate_business_support_phone( $value, $request, $param );
+		$this->assertEquals( $return, $expected );
+	}
+
+	/**
+	 * Provider for test_validate_business_support_phone.
+	 * @return array[] test method params.
+	 */
+	public function account_business_support_phone_validation_provider() {
+		$request = new WP_REST_Request();
+		return [
+			[
+				'123-123456',
+				$request,
+				'account_business_support_phone',
+				true,
+			],
+			[
+				'', // Empty value should be allowed.
+				$request,
+				'account_business_support_phone',
+				true,
+			],
+			[
+				'123test',
+				$request,
+				'account_business_support_phone',
+				new WP_Error( 'rest_invalid_pattern', 'Error: Invalid phone number: 123test' ),
+			],
+		];
+	}
+
+	/**
+	 * Tests account business support URL validator
+	 *
+	 * @dataProvider account_business_support_uri_validation_provider
+	 */
+	public function test_validate_business_support_uri( $value, $request, $param, $expected ) {
+		$return = $this->controller->validate_business_support_uri( $value, $request, $param );
+		$this->assertEquals( $return, $expected );
```

Comment:
`assertSame`?

Reply:
Originally I have tried to use `assertSame`, but asserting two objects with data provider always return false, because they are not referencing the same object.

Reply:
Maybe it's worth adding the type comparison as well. Check that value is also bool or WP_Error.